File: 1678-goal-parser-interpretation/1678-goal-parser-interpretation.java
class Solution {
    public String interpret(String command) {
        StringBuilder str = new StringBuilder();
        for(int i = 0; i < command.length(); i++){
            if(command.charAt(i) == 'G')
                str.append('G');
            else if(command.charAt(i) == '(' && command.charAt(i+1) == ')'){
                str.append('o');
                i++;
            }
            else{
                str.append('a');
                str.append('l');
                i += 3;
            }
        }
        return str.toString();
    }
}

File: 1678-goal-parser-interpretation/1678-goal-parser-interpretation.swift
class Solution {
    func interpret(_ command: String) -> String {
        var arr = [Character]()
        
        for (i,val) in command.enumerated() {
            if val == "G" {
                arr.append(val)
            } else if val == "(" {
                let index =  command.index(command.startIndex, offsetBy: i+1)
                if command[index] == "a" {
                    arr.append("a")
                    arr.append("l")
                } else {
                    arr.append("o")
                }
            }
        }
        
        return String(arr)    // Character Array ---> String
    }
}